build failed on all x32 target(including `i686`, `armv7`)

build compiled failted with:
```
error[E0308]: mismatched types
   --> /cargo/registry/src/rsproxy.cn-0dccff568467c15b/global-hotkey-0.6.1/src/platform_impl/x11/mod.rs:126:21
    |
122 |                 (xlib.XGrabKey)(
    |                 --------------- arguments to this function are incorrect
...
126 |                     root,
    |                     ^^^^ expected `u32`, found `u64`
    |
help: you can convert a `u64` to a `u32` and panic if the converted value doesn't fit
    |
126 |                     root.try_into().unwrap(),
    |                         ++++++++++++++++++++

error[E0308]: mismatched types
   --> /cargo/registry/src/rsproxy.cn-0dccff568467c15b/global-hotkey-0.6.1/src/platform_impl/x11/mod.rs:135:86
    |
135 |                     unsafe { (xlib.XUngrabKey)(display, keycode as _, modifiers | m, root) };
    |                              -----------------                                       ^^^^ expected `u32`, found `u64`
    |                              |
    |                              arguments to this function are incorrect
    |
help: you can convert a `u64` to a `u32` and panic if the converted value doesn't fit
    |
135 |                     unsafe { (xlib.XUngrabKey)(display, keycode as _, modifiers | m, root.try_into().unwrap()) };
    |                                                                                          ++++++++++++++++++++

error[E0308]: mismatched types
   --> /cargo/registry/src/rsproxy.cn-0dccff568467c15b/global-hotkey-0.6.1/src/platform_impl/x11/mod.rs:175:78
    |
175 |             unsafe { (xlib.XUngrabKey)(display, keycode as _, modifiers | m, root) };
    |                      -----------------                                       ^^^^ expected `u32`, found `u64`
    |                      |
    |                      arguments to this function are incorrect
    |
help: you can convert a `u64` to a `u32` and panic if the converted value doesn't fit
    |
175 |             unsafe { (xlib.XUngrabKey)(display, keycode as _, modifiers | m, root.try_into().unwrap()) };
    |                                                                                  ++++++++++++++++++++

error[E0308]: mismatched types
   --> /cargo/registry/src/rsproxy.cn-0dccff568467c15b/global-hotkey-0.6.1/src/platform_impl/x11/mod.rs:253:33
    |
250 | ...                   let _ = tx.send(register_hotkey(
    |                                       --------------- arguments to this function are incorrect
...
253 | ...                       root,
    |                           ^^^^ expected `u64`, found `u32`
    |
note: function defined here
   --> /cargo/registry/src/rsproxy.cn-0dccff568467c15b/global-hotkey-0.6.1/src/platform_impl/x11/mod.rs:105:4
    |
105 | fn register_hotkey(
    |    ^^^^^^^^^^^^^^^
...
108 |     root: u64,
    |     ---------
help: you can convert a `u32` to a `u64`
    |
253 |                                 root.into(),
    |                                     +++++++

error[E0308]: mismatched types
   --> /cargo/registry/src/rsproxy.cn-0dccff568467c15b/global-hotkey-0.6.1/src/platform_impl/x11/mod.rs:261:69
    |
261 | ...                   register_hotkey(&xlib, display, root, &mut hotkeys, hotkey)
    |                       ---------------                 ^^^^ expected `u64`, found `u32`
    |                       |
    |                       arguments to this function are incorrect
    |
note: function defined here
   --> /cargo/registry/src/rsproxy.cn-0dccff568467c15b/global-hotkey-0.6.1/src/platform_impl/x11/mod.rs:105:4
    |
105 | fn register_hotkey(
    |    ^^^^^^^^^^^^^^^
...
108 |     root: u64,
    |     ---------
help: you can convert a `u32` to a `u64`
    |
261 |                                     register_hotkey(&xlib, display, root.into(), &mut hotkeys, hotkey)
    |                                                                         +++++++

   Compiling os_info v3.8.2
error[E0308]: mismatched types
   --> /cargo/registry/src/rsproxy.cn-0dccff568467c15b/global-hotkey-0.6.1/src/platform_impl/x11/mod.rs:272:33
    |
269 | ...                   let _ = tx.send(unregister_hotkey(
    |                                       ----------------- arguments to this function are incorrect
...
272 | ...                       root,
    |                           ^^^^ expected `u64`, found `u32`
    |
note: function defined here
   --> /cargo/registry/src/rsproxy.cn-0dccff568467c15b/global-hotkey-0.6.1/src/platform_impl/x11/mod.rs:159:4
    |
159 | fn unregister_hotkey(
    |    ^^^^^^^^^^^^^^^^^
...
162 |     root: u64,
    |     ---------
help: you can convert a `u32` to a `u64`
    |
272 |                                 root.into(),
    |                                     +++++++

error[E0308]: mismatched types
   --> /cargo/registry/src/rsproxy.cn-0dccff568467c15b/global-hotkey-0.6.1/src/platform_impl/x11/mod.rs:280:71
    |
280 | ...                   unregister_hotkey(&xlib, display, root, &mut hotkeys, hotkey)
    |                       -----------------                 ^^^^ expected `u64`, found `u32`
    |                       |
    |                       arguments to this function are incorrect
    |
note: function defined here
   --> /cargo/registry/src/rsproxy.cn-0dccff568467c15b/global-hotkey-0.6.1/src/platform_impl/x11/mod.rs:159:4
    |
159 | fn unregister_hotkey(
    |    ^^^^^^^^^^^^^^^^^
...
162 |     root: u64,
    |     ---------
help: you can convert a `u32` to a `u64`
    |
280 |                                     unregister_hotkey(&xlib, display, root.into(), &mut hotkeys, hotkey)
error: could not compile `global-hotkey` (lib) due to 7 previous errors
```

--- a/.changes/fix-linux-32bit.md
+++ b/.changes/fix-linux-32bit.md
@@ -1,0 +1,5 @@
+---
+global-hotkey: patch
+---
+
+Fixed an issue causing compilation to fail for 32-bit targets.

--- a/src/platform_impl/x11/mod.rs
+++ b/src/platform_impl/x11/mod.rs
@@ -2,7 +2,7 @@
 // SPDX-License-Identifier: Apache-2.0
 // SPDX-License-Identifier: MIT
 
-use std::{collections::BTreeMap, ptr};
+use std::{collections::BTreeMap, ffi::c_ulong, ptr};
 
 use crossbeam_channel::{unbounded, Receiver, Sender};
 use keyboard_types::{Code, Modifiers};
@@ -105,7 +105,7 @@ const IGNORED_MODS: [u32; 4] = [
 fn register_hotkey(
     xlib: &Xlib,
     display: *mut _XDisplay,
-    root: u64,
+    root: c_ulong,
     hotkeys: &mut BTreeMap<u32, Vec<(u32, u32, bool)>>,
     hotkey: HotKey,
 ) -> crate::Result<()> {
@@ -159,7 +159,7 @@ fn register_hotkey(
 fn unregister_hotkey(
     xlib: &Xlib,
     display: *mut _XDisplay,
-    root: u64,
+    root: c_ulong,
     hotkeys: &mut BTreeMap<u32, Vec<(u32, u32, bool)>>,
     hotkey: HotKey,
 ) -> crate::Result<()> {
@@ -189,7 +189,7 @@ fn events_processor(thread_rx: Receiver<ThreadMessage>) {
     if let Ok(xlib) = xlib::Xlib::open() {
         unsafe {
             let display = (xlib.XOpenDisplay)(ptr::null());
-            let root = (xlib.XDefaultRootWindow)(display);
+            let root: c_ulong = (xlib.XDefaultRootWindow)(display);
 
             // Only trigger key release at end of repeated keys
             let mut supported_rtrn: i32 = 0;

